### DISCUSSION TITLE
Lowercase country code in language code

### DISCUSSION THREAD

**Discussion by sr258** [WeblateOrg/weblate#7450] [Q&A] [answered]
Hi,

I've set the language code style of projects to `BCP style using hyphen as a separator`. The filenames that are generated for new files are like `de-DE.json` or `pt-BR.json`. The system I target requires names with lower case country codes (`de-de.json` or `pt-br.json`). Is there a way to make Weblate user lowercase country codes? If necessary, I could commission a code change.

Best
Sebastian

**Comment by nijel** [comment, marked as answer]
There is currently no such option, see https://docs.weblate.org/en/latest/admin/languages.html#adding-new-translations for available ones. Adding lowercase option should be easy (add choice to `weblate/trans/models/component.py` and implement it in `weblate/formats/base.py`).

**Reply by sr258** [reply]
Thank you for pointing me in the right direction! Any chance of a lowercase BCP style being accepted as a PR or is this of too little general use?

**Reply by nijel** [reply]
I think it makes sense. In the long term, we might look for more generic solution to this, though...

**Reply by nijel** [reply]
I've added it while touching this code in https://github.com/WeblateOrg/weblate/pull/7901
